drivers/spi/spi_max32.c

```diff
@@ -784,22 +945,28 @@ static const struct spi_driver_api spi_max32_api = {
 #define MAX32_SPI_DMA_INIT(n)
 #endif
 
-#define DEFINE_SPI_MAX32(_num)                                                                     \
-	PINCTRL_DT_INST_DEFINE(_num);                                                              \
-	SPI_MAX32_IRQ_CONFIG_FUNC(_num)                                                            \
-	static const struct max32_spi_config max32_spi_config_##_num = {                           \
+#define DEFINE_SPI_MAX32_RTIO(_num) SPI_RTIO_DEFINE(max32_spi_rtio_##_num,                 \
+			CONFIG_SPI_MAX32_RTIO_SQ_SIZE,                              \
+			CONFIG_SPI_MAX32_RTIO_CQ_SIZE)
+
+#define DEFINE_SPI_MAX32(_num)                                                             \
+	PINCTRL_DT_INST_DEFINE(_num);                                                          \
+	SPI_MAX32_IRQ_CONFIG_FUNC(_num)                                                        \
+	COND_CODE_1(CONFIG_SPI_RTIO, (DEFINE_SPI_MAX32_RTIO(_num)), ());                       \
+	static const struct max32_spi_config max32_spi_config_##_num = {                       \
 		.regs = (mxc_spi_regs_t *)DT_INST_REG_ADDR(_num),                                  \
 		.pctrl = PINCTRL_DT_INST_DEV_CONFIG_GET(_num),                                     \
 		.clock = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(_num)),                                 \
 		.perclk.bus = DT_INST_CLOCKS_CELL(_num, offset),                                   \
 		.perclk.bit = DT_INST_CLOCKS_CELL(_num, bit),                                      \
 		MAX32_SPI_DMA_INIT(_num) SPI_MAX32_CONFIG_IRQ_FUNC(_num)};                         \
-	static struct max32_spi_data max32_spi_data_##_num = {                                     \
+	static struct max32_spi_data max32_spi_data_##_num = {                                 \
 		SPI_CONTEXT_INIT_LOCK(max32_spi_data_##_num, ctx),                                 \
 		SPI_CONTEXT_INIT_SYNC(max32_spi_data_##_num, ctx),                                 \
-		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(_num), ctx)};                          \
-	DEVICE_DT_INST_DEFINE(_num, spi_max32_init, NULL, &max32_spi_data_##_num,                  \
-			      &max32_spi_config_##_num, PRE_KERNEL_2, CONFIG_SPI_INIT_PRIORITY,    \
+		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(_num), ctx)                            \
+		IF_ENABLED(CONFIG_SPI_RTIO, (.rtio_ctx = &max32_spi_rtio_##_num))};                \
+	DEVICE_DT_INST_DEFINE(_num, spi_max32_init, NULL, &max32_spi_data_##_num,              \
+			      &max32_spi_config_##_num, PRE_KERNEL_2, CONFIG_SPI_INIT_PRIORITY,   \
```

Comment:
Some spurious whitespace changes are adding noise to the diff. Can you revert those please?

Reply:
Should be reverted now